### PR TITLE
Add support for resources of type StatefulSet

### PR DESCRIPTION
**Pull request by jbndbc**
Multiple resources either from a single yaml file or from multiple
files in a single directory can now also be handled by a single
deploy-k8s command execution.

### CODE REVIEW

File: src/kubernetes_deploy_tools/deployer.py [hunk 1]
```diff
@@ -15,37 +15,49 @@ def __init__(self, namespace, kubecfg=None):
...
-        rollout_completed = self.__rollout_completion(appname)
-        if rollout_completed:
-            print("Success")
-        else:
+        resources = self.__apply(filename, appname)
+        rollout_failed = [resource for resource in resources if not self.__rollout_completion(resource)]
+        if rollout_failed:
             print("Error - Rolling back")
-            self.__rollback_current_deployment(appname)
+            [self.__rollback(resource) for resource in rollout_failed]
```
Comment (damkjaer) [inline src/kubernetes_deploy_tools/deployer.py:47]: You only roll back resources that has failed. 
Is this always the case, or are there circumstances where you want to roll back all resources?

**Review by damkjaer — COMMENTED** [review]
You only roll back resources that has failed.
Is this always the case, or are there circumstances where you want to roll back all resources?

**Comment by jbndbc** [comment]
I think rolling back failed resources only is the lesser of two evils.

**Comment by damkjaer** [comment]
Just trying to understand:
What if I deploy both A (revision 10) and B (revision 4) and only B fails. Then B will be rolled back but A will not. If there is an internal connection between services then this may be broken? Or am I misunderstanding something?

**Comment by jbndbc** [comment]
Yes, that would be the case, and that would be bad.

But very bad things could also happen with the rollback all solution. If we also rollback the otherwise healthy service of the two, we might actually loose functionality which other services had grown dependent on.

I'm not sure there is a perfect solution for this, and I think in general these "mixed" gitops projects should be used with extreme care.

**Review by damkjaer — APPROVED** [review]
Looks good.

**Pull request merged**